src/common/options/CMakeLists.txt

```diff
@@ -74,13 +74,6 @@ if(FREEBSD)
 endif()
 
 set(mgr_disabled_modules "")
-if(WITH_MGR)
-  # https://tracker.ceph.com/issues/45147
-  if(Python3_VERSION VERSION_GREATER_EQUAL 3.8)
```

Comment:
my major concern is https://tracker.ceph.com/issues/45147 . if python3.9 is proved to be safe, we should instead make python3.8 an exception, something like
```cmake
if(Python3_VERSION VERSION_EQUAL 3.8)
```

Reply:
Hi @tchaikov 

That sounds like a reasonable precaution. To extend that thought, there is no guarantee it will work in 3.7 either. The new libraries I've updated to are recommended from 3.9 and up.

They might work just fine for 3.7- but not guaranteed.

So I see two approaches.

Either create a test harness that tests all three versions to guarantee interoperability and test 3.8 manually to see if we can reproduce the issue.

Or if we just want to exclude the version, that could also be ok when some of the distributors of supported distros made the jump over 3.8 anyway.

Best regards
Daniel

Reply:
> Hi @tchaikov

hi Daniel,

> 
> That sounds like a reasonable precaution. To extend that thought, there is no guarantee it will work in 3.7 either. The new libraries I've updated to are recommended from 3.9 and up.
> 
> They might work just fine for 3.7- but not guaranteed.
> 

agreed. we do disable [0] this mgr module for ubuntu focal, which comes with python 3.8. and it is enabled for centos 8 stream, which comes with python3.6.8.

---
[0] search "mgr module disabled for" in https://jenkins.ceph.com/job/ceph-dev-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=focal,DIST=focal,MACHINE_SIZE=gigantic/49953//consoleFull

> So I see two approaches.
> 
> Either create a test harness that tests all three versions to guarantee interoperability and test 3.8 manually to see if we can reproduce the issue.

a little more background first:

- the integration test environment we have: ubuntu focal and centos 8
- the existing test harness: `qa/tasks/mgr/test_module_selftest.py`

we only have python 3.6.8 and python3.8 for integration tests. yeah, we *could* have python3.7 as well using tox driven tests, as debian buster comes with python3.7 and Luminous (Ceph 12.2). but the question is, what distro we should test for? are we test on python3.7 just for the sake of completeness? if that's the case, i'm not sure if it's worthy doing so. and i think we should focus on current/future stable releases. they are more likely to ship python3.8 and newer.

regarding reproduce on 3.8,  please see https://tracker.ceph.com/issues/45147 , i'd avoid repeat what we did 2 years ago unless we have something new. for instance, if python had a fix of https://bugs.python.org/issue15751 for python3.8.

> 
> Or if we just want to exclude the version, that could also be ok when some of the maintainers of supported distros made the jump over 3.8 anyway.

debian stable comes with python3.9. 3.8 is much more popular than 3.7
> 
> Best regards Daniel

Reply:
Hi @tchaikov 

I've read the resources you supplied, and I've not dived deep into testing on different versions or similar, but most of the reports are talking about issues with NumPy. This PR will change the NumPy version to a newer one that should support newer versions of Python. Then again, to be safe, I see no reason why we should not exclude 3.8.

I've made a small commit changing this PR to follow your advice. 

Thank you for reviewing.

Best regards
Daniel